File: src/windows/CLK.Promises/Progress.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CLK.Promises
{
    public sealed class Progress
    {
        // Constructors
        public Progress(string localizedDescription, int completedCount, int totalCount)
        {
            #region Contracts

            if (string.IsNullOrEmpty(localizedDescription) == true) throw new ArgumentNullException();

            #endregion

            // Default           
            this.LocalizedDescription = localizedDescription;
            this.CompletedCount = completedCount;
            this.TotalCount = totalCount;            
        }


        // Properties 
        public string LocalizedDescription { get; }

        public int CompletedCount { get; }

        public int TotalCount { get; }      
    }
}

File: src/android/com.clk.promises/src/com/clk/promises/Progress.java
package com.clk.promises;

import com.clk.StringHelper;

public class Progress {
	
	// constructors
    public Progress(String localizedDescription, int completedCount, int totalCount)
    {
    	// contracts
        if (StringHelper.isNullOrEmpty(localizedDescription) == true) throw new IllegalArgumentException();
        
        // default            
    	_completedCount = completedCount;
    	_totalCount = totalCount;
    	_localizedDescription = localizedDescription;
    }


    // properties
    private String _localizedDescription;
	public final String getLocalizedDescription()
	{
		return _localizedDescription;
	}
	
    private int _completedCount;
	public final int getCompletedCount()
	{
		return _completedCount;
	}
	
	private int _totalCount;
	public final int getTotalCount()
	{
		return _totalCount;
	}
}